dev/tools/gen_defaults/test/gen_defaults_test.dart

```diff
@@ -325,10 +325,69 @@ static final String tokenBar = 'foo';
       logger.log('foobar');
       logger.printTokensUsage(verbose: true);
 
-      expect(printLog, contains(errorColoredString('Token unavailable: baz')));
-      expect(printLog, contains(errorColoredString('Token unavailable: foobar')));
       expect(printLog, contains('❌ foo'));
       expect(printLog, contains('Tokens used: 0/1'));
+      expect(printLog, contains(errorColoredString('Some referenced tokens do not exist: 2')));
+      expect(printLog, contains('  baz'));
+      expect(printLog, contains('  foobar'));
+    }));
+
+    test("color function doesn't log when providing a default", overridePrint(() {
+      allTokens['color_foo_req'] = 'value';
+
+      // color_foo_opt is not available, but because it has a default value, it won't warn about it
+
+      TestColorTemplate('block', 'filename', allTokens).generate();
+      logger.printTokensUsage(verbose: true);
+
+      expect(printLog, contains('✅ color_foo_req'));
+      expect(printLog, contains('Tokens used: 1/1'));
+    }));
+
+    test('color function logs when not providing a default', overridePrint(() {
+      // Nor color_foo_req or color_foo_opt are available, but only color_foo_req will be logged.
+      // This mimics a token being removed, but expected to exist.
+
+      TestColorTemplate('block', 'filename', allTokens).generate();
+      logger.printTokensUsage(verbose: true);
+
+      expect(printLog, contains('Tokens used: 0/0'));
+      expect(printLog, contains(errorColoredString('Some referenced tokens do not exist: 1')));
```

Comment:
Just a question. Why here we expect 1?

Reply:
It's somewhat explained in the comment below the test name. Maybe it could be reworded if it's not clear.
Basically, the test reads 2 tokens, one with a default value and the other without one. In this test, there are no token values for those token names, so there is 1 warning/error for the token that didn't provide a default value.

The output would be:

```
Some referenced tokens do not exist: 1
      color_foo_req
```

Reply:
Oh I see, thanks! I'm investigating this issue https://github.com/flutter/flutter/issues/142716, it seems always provide a 7 instead of 1.

Reply:
@QuncCccccc I was looking into it too, as it remind me of a similar error I got when I was working on that PR for gen_defaults. It seems to be because of the global logger as Zachary pointed out.
I believe this PR should fix it: https://github.com/flutter/flutter/pull/142743

Reply:
Ah nice! Thanks for the super quick action:) Will take a look at the fix!